ui/base_classes.py

```diff
@@ -0,0 +1,49 @@
+import bpy
+from bpy.props import *
+from bpy.types import Panel
+from ..resources.icons.icons import load_icons
+
+# Standard panel
+class PRManPanel(Panel):
+    bl_space_type = "PROPERTIES"
+    bl_region_type = "WINDOW"
+    bl_context = "render"
+
+    @classmethod
+    def poll(cls, context):
+        rd = context.scene.render
+        return rd.engine == 'PRMAN_RENDER'
+
+
+# Panel used for any collections (mesh, layers)
+class PRManCollectionPanel(PRManPanel):
+
+    def _draw_collection(self, context, layout, ptr, name, operator,
+                         opcontext, prop_coll, collection_index, default_name=''):
+        layout.label(name)
+        row = layout.row()
+        row.template_list("UI_UL_list", "PRMAN", ptr, prop_coll, ptr,
+                          collection_index, rows=1)
+        col = row.column(align=True)
+
+        op = col.operator(operator, icon="ZOOMIN", text="")
+        op.context = opcontext
+        op.collection = prop_coll
+        op.collection_index = collection_index
+        op.defaultname = default_name
+        op.action = 'ADD'
+
+        op = col.operator(operator, icon="ZOOMOUT", text="")
+        op.context = opcontext
+        op.collection = prop_coll
+        op.collection_index = collection_index
+        op.action = 'REMOVE'
+
+        if hasattr(ptr, prop_coll) and len(getattr(ptr, prop_coll)) > 0 and \
+                getattr(ptr, collection_index) >= 0:
+            item = getattr(ptr, prop_coll)[getattr(ptr, collection_index)]
+            self.draw_item(layout, context, item)
```

Comment:
I see what you're saying but I'm not sure how to code it properly given that the item parameters are going to change for every ui panel it appears in.